### ISSUE TITLE
Campaign still appears to have syncing problems in multiplayer

### ISSUE THREAD

**Issue by NilanthAnimosus** [Regalis11/Barotrauma#484]
Version 0.8.1.4 - Vanilla

I attempted a new game with a torrent submarine, I have not done much testing yet.

I got a vanilla 0.8.1.4 copy of the game, copied that game and ran each side by side, one as client and one as server.

The server used the client to host. additionally I setup the campaign before the client joined.

after the client joined I set to purchase a lot of the items until around 45 credits of 10000 left (Nuclear shells, railgun shells, crowbars and other random things).

I chose an explosive delivery mission - turned out this one had the cargo that exploads.

The clients did not sync and failed their initial sync (no disconnect, but clearly network errors) for an unknown reason:

Error on client side:
![](http://puu.sh/AXGIh/d5cc502540.jpg)

A comparison of both client and server (Their cargo does not match - different count of boxes such as the host carrying a box that does not exist, amongst other issues):
![](https://puu.sh/AXGUn/10baf94c35.png)

In short however, the master branch of the game appears to still have problems with sync on campaign mode. I'll reply back here if I work out any reproduction methods/steps or additional files.

reloading the campaign simply syncs correctly it seems, with the items purchased saved into the file.

Personally I don't know how to make heads or tails of the crashreport.bin files since without knowing what the ID's / events were for I have no way to gauge what it may have been for anymore, but I'll include this:

[crashreport_object.zip](https://github.com/Regalis11/Barotrauma/files/2200208/crashreport_object.zip)

**Comment by NilanthAnimosus** [comment]
Ok. I had it occur again. will likely edit this or reply again with reproduction files if possible, it seems reproducable but gives no networking errors (but I don't think my hosts cargobox is meant to be velonacepts calyx eggs so I think this will do)

It appears the mission cargo can often desync. Perhaps from the items before it having incorrect ID's but there is definately ID missmatching going on between the client and server.

Problem is simply typing "dumpids" doesn't show me anything. the final IDs shown on the client/server are
